lib/fluent/plugin/in_wmi.rb

```diff
@@ -12,18 +12,30 @@ class WmiInput < Fluent::Plugin::Input
     desc "The value is the tag assigned to the generated events."
     config_param :tag, :string
     desc "The value is the instance name of WMI."
-    config_param :class_name, :string
+    config_param :class_name, :string, default: nil
+    desc "The value is the query of WMI."
+    config_param :query, :string, default: nil
     desc "Determine the rate to emit metrics as events."
     config_param :interval, :time, default: 10
 
     def configure(conf)
       super
-      raise Fluent::ConfigError, "class_name is required." unless @class_name
+      if @class_name && @query
+        raise Fluent::ConfigError, "Cannot specify both parameters 'class_name' and 'query'."
+      end
+      if !@class_name && !@query
+        raise Fluent::ConfigError, "'class_name' or 'query' parameter is required."
+      end
       begin
-        WmiLite::Wmi.new.instances_of(@class_name)
+        WmiLite::Wmi.new.instances_of(@class_name) if @class_name
       rescue WmiLite::WmiException
         raise Fluent::ConfigError, "#{@class_name} is invalid class_name."
       end
+      begin
+        WmiLite::Wmi.new.query(@query) if @query
```

Comment:
These comments are just my curiosity. Feel free to ignore this.

These line implementation seems insufficient for me.
If we can create `@wmi` instance in `#configure` and use `@wmi` instance in the latter implementation instead of wmi object creation in every time, it would be more sufficient implementation.

Reply:
Your concern is justified.
However, the lifetime of the `Wmi` object is currently unknown to me, so if the instance becomes unusable during fluentd's life cycle, it needs to be regenerated. It's not simple.

Reply:
Ah, right.
It seems that `Wmi` object is just an wrapper for `WIN32OLE` object.
ref: https://github.com/chef/wmi-lite/blob/master/lib/wmi-lite/wmi.rb

Yeah, I have no idea about lifeime of `Wmi` object because WIN32OLE is also wrapper (binding?) for COM object....